lbccore-kmp/src/commonMain/kotlin/studio/lunabee/compose/core/kmp/LbcTextSpec.kt

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2024 Lunabee Studio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * LbcTextSpec.kt
+ * Lunabee Compose
+ *
+ * Created by Lunabee Studio / Date - 11/19/2024 - for the Lunabee Compose library.
+ */
+
+package studio.lunabee.compose.core.kmp
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.text.AnnotatedString
+import org.jetbrains.compose.resources.ExperimentalResourceApi
+import org.jetbrains.compose.resources.PluralStringResource
+import org.jetbrains.compose.resources.pluralStringResource
+import org.jetbrains.compose.resources.stringResource
+import studio.lunabee.compose.core.kmp.ext.format
+import kotlin.collections.contentEquals
+import kotlin.collections.contentHashCode
+import kotlin.collections.isEmpty
+
+@Stable
+sealed class LbcTextSpec {
+    abstract val annotated: AnnotatedString
+        @Composable
+        get
+
+    abstract val string: String
+        @Composable
+        get
+
+    @Composable
+    protected fun Array<out Any>.resolveArgs(): Array<out Any> {
+        return Array(this.size) { idx ->
+            val entry = this[idx]
+            if (entry is LbcTextSpec) {
+                entry.string // marked as compile error due to Java(?)
+            } else {
+                this[idx]
+            }
+        }
+    }
+
+    class Raw(
+        private val value: String,
+        private vararg val args: Any,
+    ) : LbcTextSpec() {
+        override val annotated: AnnotatedString
+            @Composable
+            get() = AnnotatedString(string)
+        override val string: String
+            @Composable
+            get() = if (args.isEmpty()) {
+                value
+            } else {
+                value.format(*args.resolveArgs())
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/LunabeeStudio/Lunabee_Compose_Android/blob/ba46bf847537102988e608c48a1f8d79260b8be2/lbccore-kmp/src/commonMain/kotlin/studio/lunabee/compose/core/kmp/LbcTextSpec.kt#L70"></span>In most cases using a spread operator causes a full copy of the array to be created before calling a method. This may result in a performance penalty.</td>
    </tr>
  </tbody>
</table>